[Mobile] Update media placeholders and block outline

geriux opened the pull request:
**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5969
- https://github.com/wordpress-mobile/WordPress-iOS/pull/21119
- https://github.com/wordpress-mobile/WordPress-Android/pull/18788

## What?
This PR brings back the BlockOutline that was removed in https://github.com/WordPress/gutenberg/pull/51222 and also updates the MediaPlaceholder with new styles.

## Why?
To prevent confusion of selected states of blocks and also to improve our current editor UI.

## How?
It restores the `BlockOutline` component and updates its logic so the outline will be shown when the component is selected and meets one of the following conditions: it is a `text` block with inner blocks, it is a block other than `text` with inner blocks and is at root list level, or it is a block other than `text` with inner blocks and is not at root list level.

New colors were added `$blue-60`, `$blue-40` and `$dark-dim`.

## MediaPlaceholder

Updates the component so it can be selected before the whole container had a touch action so it wouldn't be possible to be in a selected state.

Now a button will be shown in the middle to allow users to upload media, it also updates the styles and cleans up the code a bit.

## Embed Placeholder

It brings the same style modifications from the MediaPlaceholder component to this one.

## Tests

General snapshot updates and integration tests were updated after these changes.

## Testing Instructions

## Test case 1
- Open the editor
- Add several media blocks and check the placeholders can be selected and have the new styles (Image, Gallery, Cover, Embed, VideoPress)

## Test case 2
- Add a List block
- Add several list items
- Tap the "Up" arrow to select the parent block
- **Expect** to see the blue outline

## Test case 3

- Add different kinds of blocks
- Tap on them and **expect** to see the block outlines for parent blocks that have inner blocks, or blocks that are not text-based.

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast <!-- if applicable -->
iOS|Android
-|-
<video src="https://github.com/WordPress/gutenberg/assets/4885740/c43f5d5e-78a1-46ff-a26c-f81f54dae094" width=200 />|<video src="https://github.com/WordPress/gutenberg/assets/4885740/b71d3dd4-ef57-4641-b4fc-bd124c107099" width=200 />

Commit:
Add new colors: $blue-60, $blue-40 and $dark-dim

Commit:
Gallery block - Remove native icon variant

Commit:
Media Placeholder - Update colors

Commit:
Embed - Update placeholder styles

Commit:
Embed - Placeholder - Add button styles for the more options action

Commit:
Add back BlockOutline with update logic around parent blocks and media blocks

Commit:
BlockOutline - Update paddings for text blocks

Commit:
ButtonBlockAppender - Update left and right margins

Commit:
Update colors for media placeholders

Commit:
Image component - Update selected styles

Commit:
Merge branch 'trunk' into rnmobile/update-media-placeholders-and-block-outline

Commit:
MediaPlaceholder - Fix missing touch action for the appender

Commit:
Update styleMock

Commit:
Audio and File block - Update component snapshot

Commit:
Update Embed, Gallery and Image tests to use new text placeholders

SiobhyB commented on packages/block-library/src/embed/styles.native.scss:
```diff
@@ -4,43 +4,67 @@
 	align-items: center;
 	justify-content: center;
-	background-color: $gray-lighten-30;
+	background-color: #e0e0e0; // $light-dim
```
Is there anything preventing us directly using `$light-dim` here and `$dark-dim` directly on line 19? 🤔

geriux replied:
Hi! Good question! The issue is that `$light-dim` has an alpha value, so depending on the background it would show differently, so for this case, since it's the placeholder's container, we are using a solid color that's similar to `$light-dim` having it on top of a white background color.

SiobhyB replied:
Got it, thanks for clarifying!

SiobhyB commented on packages/block-editor/src/components/media-placeholder/styles.native.scss:
```diff
@@ -1,50 +1,63 @@
 	align-items: center;
 	justify-content: center;
-	background-color: $gray-lighten-30;
+	background-color: #e0e0e0; // $light-dim
```
Similar to the question added to the [embed/styles.native.scss](https://github.com/WordPress/gutenberg/pull/52702/files#diff-ca04c8499686eab5772ec24f45baac1d7b0f2f6eff67d502fb4848a4581e61b2) file, I'm curious why we'd use the hex values as opposed to using the `$light-dim` or `$dark-dim` variables in this file.

geriux replied:
Similar to my reply above, the issue is that `$light-dim` has an alpha value, so depending on the background it would show differently, so for this case, since it's the placeholder's container, we are using a solid color that's similar to `$light-dim` having it on top of a white background color.

SiobhyB reviewed (approved):
@geriux, LGTM! All test cases worked as expected in my testing. :) I only added a couple of questions to the CSS files, but those spawn from curiosity and aren't anything blocking. 

Thanks for the work on this! 🎉 👏

Commit:
Update block outline component to show border for unsupported blocks

Commit:
Synced patterns - Fix seperator line to overflow outside the block's container

Commit:
Unsupported block - Update placeholder styles

Commit:
E2E Test - Remove unsupported block test in favor of a visual test

Commit:
Missing block - Update snapshot

geriux commented:
Hey @SiobhyB 👋 thanks for reviewing and testing! I've added a couple of commits after your initial review:

- Update block outline component to show border for unsupported blocks https://github.com/WordPress/gutenberg/pull/52702/commits/f298e5847f3eaf2b029c3c446f11fdab43219e4a
- Synced patterns - Fix separator line to overflow outside the block's container https://github.com/WordPress/gutenberg/pull/52702/commits/eeeee97d69785210862320b565d32138dc00d8c3
- Unsupported block - Update placeholder styles https://github.com/WordPress/gutenberg/pull/52702/commits/2209870d92350f49b3698e8aa00c3c0f4ce8a600
- E2E Test - Remove unsupported block test in favor of a visual test https://github.com/WordPress/gutenberg/pull/52702/commits/38b9a3fde1209cf7a464818cbff5d925cd1552b2 I added a visual E2E tests for this block so I moved the only test we had to Gutenberg mobile in https://github.com/wordpress-mobile/gutenberg-mobile/pull/5969/commits/8537631bfb33e75104bbf4a4c6923eec95e8aa70

SiobhyB commented:
@geriux, perfect, thank you for following up! I reviewed and tested the most recent changes :shipit: 🚀
